src/core/services/eth-service/index.ts

```diff
@@ -39,18 +39,34 @@ export function fetchHints(info: GraphQLResolveInfo): FetchHints {
     s => s.kind === 'Field' && TX_REQUIRING_FIELDS.indexOf(s.name.value) >= 0,
   );
 
-  const logsFields = txFields.filter(
-    f => f.kind === 'Field' && f.selectionSet.selections.some(f => f.kind === 'Field' && f.name.value === 'logs'),
+  const logsFields = _.flatMap(txFields, f =>
+    (f as FieldNode).selectionSet.selections.filter(f => f.kind === 'Field' && f.name.value === 'logs'),
   );
 
+  // Extract log filters out of the GraphQLResolveInfo
+  // Since logs are resolved with blocks when they are nested in the query, we must pull out
+  // the filter arguments here so that we can use them in the blocks resolution
+  // There has to be a better way to do this...
```

Comment:
Yeah, this is somewhat evil. IMO, there are two ways of achieving this:
1. Eagerly fetching the filtered logs by running a query on the block level and merging the results into the object tree; therefore the logs will already be populated in the obj and the resolver can just return the property values.
2. Relying on the dataloader cache to dedup the backend call. Instead of resolving at the block level, we allow GraphQL to call JSON-RPC `eth_getLogs` as many times as transactions, knowing that the dataloader will only allow the first invocation through, and will park the subsequent ones, cloning the result from the call into the parked ones once it arrives.

The complexity herein comes from choosing 1. It might be worth experimenting with 2. [We already support this style of request-scoped caching](https://github.com/ConsenSys/ethql/blob/master/src/core/services/web3/index.ts#L128) so it should _really_ be transparent.